Source/Swig/scanner.c

```diff
@@ -209,26 +209,26 @@ void Scanner_idstart(Scanner *s, const char *id) {
 /* -----------------------------------------------------------------------------
  * nextchar()
  * 
- * Returns the next character from the scanner or 0 if end of the string.
+ * Returns the next character from the scanner or EOF if end of the string.
```

Comment:
It's needed to handle a literal string such as `"abc\0def"` correctly - currently SWIG gets confused (from memory I think it currently replaces all characters after a zero byte with zero bytes, so you get the same as for `"abc\0\0\0\0"`; it may actually replace them with uninitialised bytes which just happen to be zero because the string typically gets allocated from new heap space).  Such literal strings are uncommon, but they're valid C/C++ and it's not good to quietly corrupt values even in uncommon cases.

Looks like I didn't add a test for this case.  It may be hard to do cross-target-language as some just can't represent such strings, but I'll have a go.

Reply:
Actually I was getting mixed up - the point here is that `nextchar()` returns the next character from the scanner, but characters can have values 0 to 255 inclusive so we shouldn't use 0 to signal end of string.  While character value 0 is unlikely, `EOF` is what we use elsewhere to signal such things so it's more consistent to use it here too.

So I think it's an improvement to the code, but I also agree it's not needed for the rest of the changes to work so I can split it out if you prefer.

Reply:
After a little more reflection I've backed this part out (in c8f4106cdd940a07deea38e9b00f24a212f85679) - this change is complicated enough without lumping in things that don't actually need to be part of it.